File: compatibility-tests/src/test/java/nnl/rocks/kactoos/io/UncheckedOutputTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.io;

import org.junit.Test;

import java.io.IOException;

/**
 * Test case for {@link UncheckedOutput}.
 *
 * @author Alexey Semenyuk (semenyukalexey88@gmail.com)
 * @version $Id: 51441c1fb0a239d1d0cf66c30a42f9c0cc1e516d $
 * @since 0.11
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class UncheckedOutputTest {

    @Test(expected = RuntimeException.class)
    public void rethrowsCheckedToUncheckedException() {
        new UncheckedOutput(
            () -> {
                throw new IOException("intended");
            }
        ).stream();
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/io/UncheckedOutputTest.kt

package nnl.rocks.kactoos.io

import org.junit.Test

import java.io.IOException

/**
 * Test case for [UncheckedOutput].
 *
 * @author Alexey Semenyuk (semenyukalexey88@gmail.com)
 * @version $Id: 51441c1fb0a239d1d0cf66c30a42f9c0cc1e516d $
 * @since 0.11
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class UncheckedOutputTest {

    @Test(expected = RuntimeException::class)
    fun rethrowsCheckedToUncheckedException() {
        UncheckedOutput(
                { throw IOException("intended") }
        ).stream()
    }
}
